src/test/java/org/minions/devfund/bruno/SquarelotronTest.java

```diff
@@ -0,0 +1,180 @@
+        final int actualSize = square.getSize();
+        final int expectedSize = 4;
+        assertTrue(squarelotron[ROW_ZERO][COL_ZERO] == 1);
+        assertTrue(squarelotron[ROW_ZERO][COL_ONE] == 2);
+        assertTrue(squarelotron[ROW_ZERO][COL_TWO] == THREE);
+        assertTrue(squarelotron[ROW_ZERO][COL_THREE] == FOUR);
+        assertTrue(squarelotron[ROW_ONE][COL_ZERO] == FIVE);
+        assertTrue(squarelotron[ROW_ONE][COL_ONE] == SIX);
+        assertTrue(squarelotron[ROW_ONE][COL_TWO] == SEVEN);
+        assertTrue(squarelotron[ROW_ONE][COL_THREE] == EIGHT);
+        assertTrue(squarelotron[ROW_TWO][COL_ZERO] == NINE);
+        assertTrue(squarelotron[ROW_TWO][COL_ONE] == TEN);
+        assertTrue(squarelotron[ROW_TWO][COL_TWO] == ELEVEN);
+        assertTrue(squarelotron[ROW_TWO][COL_THREE] == TWELVE);
+        assertTrue(squarelotron[ROW_THREE][COL_ZERO] == THIRTEEN);
+        assertTrue(squarelotron[ROW_THREE][COL_ONE] == FOURTEEN);
+        assertTrue(squarelotron[ROW_THREE][COL_TWO] == FIFTEEN);
+        assertTrue(squarelotron[ROW_THREE][COL_THREE] == SIXTEEN);
+        assertTrue(actualSize == expectedSize);
+    }
+
+    /**
+     * Tests if the new squarelotron object makes the moves correctly.
+     */
+    @Test
+    public void testUpsideDownFlip() {
+        final int[][] squarelotronOrigin = square.getSquarelotron();
+        final Squarelotron firstFlip = square.upsideDownFlip(1);
+        final int[][] firstSquarelotron = firstFlip.getSquarelotron();
+        final Squarelotron secondFlip = firstFlip.upsideDownFlip(1);
+        final int[][] secondSquarelotron = secondFlip.getSquarelotron();
+        assertTrue(firstSquarelotron[ROW_ZERO][COL_ZERO] == THIRTEEN);
+        assertTrue(firstSquarelotron[ROW_ZERO][COL_ONE] == FOURTEEN);
+        assertTrue(firstSquarelotron[ROW_ZERO][COL_TWO] == FIFTEEN);
+        assertTrue(firstSquarelotron[ROW_ZERO][COL_THREE] == SIXTEEN);
+        assertTrue(firstSquarelotron[ROW_ONE][COL_ZERO] == NINE);
+        assertTrue(firstSquarelotron[ROW_ONE][COL_ONE] == SIX);
+        assertTrue(firstSquarelotron[ROW_ONE][COL_TWO] == SEVEN);
+        assertTrue(firstSquarelotron[ROW_ONE][COL_THREE] == TWELVE);
+        assertTrue(firstSquarelotron[ROW_TWO][COL_ZERO] == FIVE);
+        assertTrue(firstSquarelotron[ROW_TWO][COL_ONE] == TEN);
+        assertTrue(firstSquarelotron[ROW_TWO][COL_TWO] == ELEVEN);
+        assertTrue(firstSquarelotron[ROW_TWO][COL_THREE] == EIGHT);
+        assertTrue(firstSquarelotron[ROW_THREE][COL_ZERO] == 1);
+        assertTrue(firstSquarelotron[ROW_THREE][COL_ONE] == 2);
+        assertTrue(firstSquarelotron[ROW_THREE][COL_TWO] == THREE);
+        assertTrue(firstSquarelotron[ROW_THREE][COL_THREE] == FOUR);
+        assertTrue(squarelotronOrigin[ROW_ZERO][COL_ZERO] == secondSquarelotron[ROW_ZERO][COL_ZERO]);
+        assertTrue(squarelotronOrigin[ROW_TWO][COL_TWO] == secondSquarelotron[ROW_TWO][COL_TWO]);
```

Comment:
In order to have more understandable unit tests.
You can write your unit tests like this.
```java
    @Test
    public void testUpsideDownFlipDimension4Ring1() {
        final int ringNumber = 1;
        final int dimension = 4;
        final Squarelotron squarelotron = new Squarelotron(dimension);
        final int[][] expectedMatrix = new int[][]{{13, 14, 15, 16}, {9, 6, 7, 12}, {5, 10, 11, 8}, {1, 2, 3, 4}};
        Squarelotron result = squarelotron.upsideDownFlip(ringNumber);
        assertArrayEquals(expectedMatrix, result.getCopySquarelotron());
    }
```